File: test-drive-xml-main-runner/src/test/java/com/steelzack/test/drive/books/XmlTestRunnerWorkerTest.java
package com.steelzack.test.drive.books;

import static org.junit.Assert.assertEquals;

import java.io.FileNotFoundException;
import java.io.InputStream;
import java.util.Random;

import org.junit.Test;

public class XmlTestRunnerWorkerTest {
	private static final Random random = new Random();

	@Test
	public void testXmlTestRunnerWorker() throws Exception {
		final String testFile = "testFile" + Math.random();
		final String testIteration = "" + random.nextInt();
		final String[] args = new String[] { //
				"-n", //
				testIteration, //
				"-b", //
				testFile, //
				"--noop" //
		};

		final XmlTestRunnerWorker runner = new XmlTestRunnerWorker(args);

		assertEquals(testIteration, runner.getnIterations().toString());
		assertEquals(testFile, runner.getXmlFile());
	}

	@Test
	public void testRunTests_10_Iter() throws Exception {
		final XmlTestRunnerWorker runner = new XmlTestRunnerWorker() {
			@Override
			InputStream getInputStreamFromFile() throws FileNotFoundException {
				return getClass().getResourceAsStream("testXmlFile.xml");
			}

			@Override
			public Integer getnIterations() {
				return 10;
			}
		};
		runner.runTests();
	}

	@Test
	public void testRunTests_10000_Iter() throws Exception {
		final XmlTestRunnerWorker runner = new XmlTestRunnerWorker() {
			@Override
			InputStream getInputStreamFromFile() throws FileNotFoundException {
				return getClass().getResourceAsStream("testXmlFile.xml");
			}

			@Override
			public Integer getnIterations() {
				return 10000;
			}
		};
		runner.runTests();
	}
}

File: xml-performance/test-drive-xml-main-runner/src/test/kotlin/org/jesperancinha/test/drive/books/XmlTestRunnerWorkerTest.kt
package org.jesperancinha.test.drive.books

import io.kotest.matchers.nulls.shouldNotBeNull
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import picocli.CommandLine
import java.io.FileNotFoundException
import java.io.InputStream
import java.util.*
import kotlin.io.path.toPath

class XmlTestRunnerWorkerTest {
    @Test
    @Throws(Exception::class)
    fun testXmlTestRunnerWorker() {
        val testFile = "testFile" + Math.random()
        val testIteration = "" + random.nextInt()
        val args = arrayOf(
            "-n",
            testIteration,
            "-b",
            testFile,
            "--noop"
        )
        val xmlTestRunner = XmlTestRunner()
        CommandLine(xmlTestRunner).execute(*args)
        Assertions.assertEquals(testIteration, xmlTestRunner.numberOfIterations.toString())
        Assertions.assertEquals(testFile, xmlTestRunner.bookTestFile)
    }

    @Test
    @Throws(Exception::class)
    fun testRunTests_10_Iter() {
        val runner: XmlTestRunner = object : XmlTestRunner() {
            override var bookTestFile: String? =
                javaClass.getResource("testXmlFile.xml").shouldNotBeNull().toURI().toPath().toString()
            override var numberOfIterations: Int? = 10
        }
        runner.runTests()
    }

    @Test
    @Throws(Exception::class)
    fun testRunTests_10000_Iter() {
        val runner: XmlTestRunner = object : XmlTestRunner() {
            override var bookTestFile: String? =
                javaClass.getResource("testXmlFile.xml").shouldNotBeNull().toURI().toPath().toString()
            override var numberOfIterations: Int? = 10000
        }
        runner.runTests()
    }

    companion object {
        private val random = Random()
    }
}